hazelcast/src/main/java/com/hazelcast/query/impl/Indexes.java

```diff
@@ -61,6 +62,8 @@ public class Indexes {
     private static final ThreadLocal<CachedQueryEntry[]> CACHED_ENTRIES =
             ThreadLocal.withInitial(() -> new CachedQueryEntry[]{new CachedQueryEntry(), new CachedQueryEntry()});
 
+    private final Node node;
+    private final String mapName;
```

Comment:
why do we need `mapName`? I expect to keep `Indexes` isolated from `IMap` or any data structure. This introduces technical debt.

Reply:
Unfortunately we introduce this dependency in the tstore b+tree indexes because to get a value from the index we have to identify `Index` object that depends on the `mapName/partitionID`.